jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1251,10 +1251,6 @@ public String getRemoteUser()
     @Override
     public RequestDispatcher getRequestDispatcher(String path)
     {
-        // path is encoded, potentially with query
-
-        path = URIUtil.compactPath(path);
-
```

Comment:
@joakime Can you comment on why this was added?

I think removing it is safe, as it may mean that an application might not find a dispatcher that it previously did, but it should never result in a dispatcher being found that should not be???

Reply:
This was added as a result of issue https://bugs.eclipse.org/482056 into Jetty 8.1.18 back in 2015.

IIRC, it was because people were using it lazily, like ...

``` java
final String DATAPATH = "/data/controller.do";
request.getRequestDispatcher("/app/" + DATAPATH) 
```

which results in `getRequestDispatcher("/app//data/controller.do")` style of usage.

or 

``` java
request.getRequestDispatcher(request.getContextPath() + "/data/foo");
```

which can easily result in a double slash as well (when we used to return `/` on root context).

I'm good with changing this for Jetty 10+, being a bit more strict in the process.
We should probably make this a configurable switch though.

Reply:
If need be, you can always have a filter that is on FORWARD only that will look for // and do a new dispatch to the compacted path.